Passing millis as part of Message for date parsing

bwtril-justin opened the pull request:
The current implementation of the message parser discards milliseconds from messages when present (e.g. the message in the [`test_empty_sd_value`](https://github.com/Roguelazer/rust-syslog-rfc5424/blob/d580edfc5ba53abcb185f51f05a5fdc5485c0f19/src/parser.rs#L495) test).  This PR attempts to preserve and pass this information as part of the Message struct via a new `timestamp_millis` value.

Roguelazer commented on src/parser.rs:
```diff
@@ -241,7 +243,7 @@ fn parse_timestamp(m: &str) -> ParseResult<(Option<time_t>, &str)> {
     if rest.starts_with('.') {
         take_char!(rest, '.');
-        take_item!(parse_num(rest, 1, 6), rest);
+        tm.tm_nsec = take_item!(parse_num(rest, 1, 6), rest) * NSEC_PER_MS;
```
technically this can have up to microsecond precision, right? so you need to look at how many digits you actually got back...

Roguelazer commented on src/parser.rs:
```diff
@@ -332,7 +334,8 @@ fn parse_message_s(m: &str) -> ParseResult<SyslogMessage> {
        version,
-       timestamp,
+       timestamp: event_time.map(|t| t.sec),
+       timestamp_millis: event_time.map(|t| t.nsec / NSEC_PER_MS),
```
I'm not sure I love having a division in the critical path here; might be better to just natively expose nanos

bwtril-justin commented:
@Roguelazer I apologize for all the broken builds; I keep neglecting to run tests with the `--all-features` flag when testing locally, and failed to notice the broken CI build prior to today.

I've hopefully addressed your concerns with the latest changes; I would appreciate any feedback if you don't care for this approach or have other suggestions.

Roguelazer commented:
Thanks for making those changes! This looks great! I'm going to do some quick integration tests on my side.

Roguelazer commented:
This is now available as v0.6.0
